Prevent the exposure of song paths for regular users

rochakjain361 opened the pull request:
Referring to: https://github.com/navidrome/navidrome/issues/959
I have fixed the UI to show the song `path` in the song details.
For the admin:

![Screenshot from 2021-04-03 15-36-32](https://user-images.githubusercontent.com/57208018/113475477-940e4a80-9493-11eb-8ade-91e175e76314.png)

For the regular users:

![Screenshot from 2021-04-03 15-37-12](https://user-images.githubusercontent.com/57208018/113475478-953f7780-9493-11eb-984f-adb254f59dc1.png)

jvoisin commented:
The original issue, #959, says:

> For security reasons, paths should not be exposed to users that are not administrators.

Since the check is implemented client-side in this pull-request, it doesn't really address the issue.

rochakjain361 commented:
@jvoisin Actually the `Writable.js` and some other components retrieved the admin role from the local storage of the browser, so earlier I implemented it on the client-side (which I wasn't aware that it was wrong, as the regular user can alter the local storage to get the path) but now I've changed the code and now it is using permissions from react-admin so now you can take a look.

deluan commented:
Hey @rochakjain361 , what @jvoisin is saying is that, to really address the issue, we need to change the backend to not send the path in the response for the `/app/api/song` endpoint, or else the path can still be seen in the "inspect" panel, or querying the API directly.

Even though I said in the issue "We should at least remove the path from the UI", for this to be complete we need to also change the backend to not send the path.

rochakjain361 commented:
@deluan, actually I was working on the backend for this issue, I forgot to mention so :sweat_smile:. I'll try to find some solution and add the work as soon as possible
